agent/rpc/middleware/interceptors.go

```diff
@@ -18,16 +18,18 @@ import (
 // Really what we are measuring here is a "cluster operation". The term we have
 // used for this historically is "RPC", so we continue to use that here.
 const RPCTypeInternal = "internal"
+const RPCTypeNetRPC = "net/rpc"
 
 var metricRPCRequest = []string{"rpc", "server", "request"}
 var requestLogName = "rpc.server.request"
 
 type RequestRecorder struct {
-	Logger hclog.Logger
+	Logger       hclog.Logger
+	recorderFunc func(key []string, start time.Time, labels []metrics.Label)
 }
 
 func NewRequestRecorder(logger hclog.Logger) *RequestRecorder {
-	return &RequestRecorder{Logger: logger}
+	return &RequestRecorder{Logger: logger, recorderFunc: metrics.MeasureSinceWithLabels}
```

Comment:
Wondering if we should not inject this to the ctor to be able to use the `recorderStore` at a higher level (Server) and use the same technique of mocking to test at higher level too.

Reply:
Oh I see -- we could pass it a "mock"/ fake recordFunc in some `agent_test` or `server_test` use cases 🤔 .

I think that way, couple with the basic testing in `metrics_test` should be enough to gain reasonable confidence and make sure there's no regressions.